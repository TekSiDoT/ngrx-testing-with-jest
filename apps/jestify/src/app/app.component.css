#addJester {
  position: fixed;
  left: 20px;
  top: 10px;
}

#partyHard {
  position: fixed;
  left: 100px;
  top: 10px;
}

#killParty {
  position: fixed;
  left: 187px;
  top: 10px;
}
