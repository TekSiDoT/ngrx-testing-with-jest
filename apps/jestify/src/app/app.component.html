<div [class.disco]="appFacade.partyHard$ | async"></div>
<button id="addJester" type="button" (click)="appFacade.setJesters(jesters.length + 1)">Add Jester</button>
<button id="partyHard" type="button" (click)="appFacade.partyHard()">Party HARD</button>
<button id="killParty" type="button" (click)="appFacade.killParty()">Kill Party</button>
<jestify-jester *ngFor="let jest of jesters"></jestify-jester>
