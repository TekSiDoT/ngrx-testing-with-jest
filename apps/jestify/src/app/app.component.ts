import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
import { AppFacade } from './+state/app.facade';
import { CookieService } from 'ngx-cookie-service';
import { catchError } from 'rxjs/operators';

export interface Coordinate {
  x: number;
  y: number;
}

@Component({
  selector: 'jestify-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AppComponent implements OnInit {
  jesters: number[] = [];
  constructor(public appFacade: AppFacade, private cookieService: CookieService) {}
  ngOnInit() {
    let jesters = 0;
    try {
      if (this.cookieService.check('jesters')) {
        jesters = parseInt(this.cookieService.get('jesters'), 10);
      }
    } catch (e) {
      console.log(e);
    } finally {
      this.appFacade.setJesters(jesters);
    }

    this.appFacade.jesters$.subscribe(cnt => (this.jesters = new Array(cnt)));
  }
}
