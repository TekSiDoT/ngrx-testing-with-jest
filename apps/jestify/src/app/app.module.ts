import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppComponent } from './app.component';
import { NxModule } from '@nrwl/nx';
import { JesterComponent } from './jester/jester.component';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { APP_FEATURE_KEY, initialState as appInitialState, appReducer } from './+state/app.reducer';
import { AppEffects } from './+state/app.effects';
import { AppFacade } from './+state/app.facade';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { environment } from '../environments/environment';
import { storeFreeze } from 'ngrx-store-freeze';
import { SafePipeModule } from 'safe-pipe';
import { CookieService } from 'ngx-cookie-service';

@NgModule({
  declarations: [AppComponent, JesterComponent],
  imports: [
    BrowserModule,
    SafePipeModule,
    NxModule.forRoot(),
    StoreModule.forRoot(
      { app: appReducer },
      {
        initialState: { app: appInitialState },
        metaReducers: !environment.production ? [storeFreeze] : []
      }
    ),
    EffectsModule.forRoot([AppEffects]),
    !environment.production ? StoreDevtoolsModule.instrument() : []
  ],
  providers: [AppFacade, CookieService],
  bootstrap: [AppComponent]
})
export class AppModule {}
