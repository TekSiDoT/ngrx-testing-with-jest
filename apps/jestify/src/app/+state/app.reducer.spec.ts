import { PartyHard, SetJesters } from './app.actions';
import { AppState, initialState, appReducer } from './app.reducer';

describe('App Reducer', () => {
  beforeEach(() => {});

  describe('valid App actions ', () => {
    it('should properly set jesters', () => {
      const action = new SetJesters(1);
      const result = appReducer(initialState, action);
      expect(result).toMatchSnapshot();
    });

    it('should properly set MANY jesters', () => {
      const action = new SetJesters(1000);
      const result = appReducer(initialState, action);
      expect(result).toMatchSnapshot();
    });

    it('should properly party', () => {
      const action = new PartyHard();
      const result = appReducer(initialState, action);
      const expectedResult: AppState = {
        partyHard: true,
        jesters: 200
      };
      expect(result).toEqual(expectedResult);
    });
  });

  describe('unknown action', () => {
    it('should return the initial state', () => {
      const action = {} as any;
      const result = appReducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });
});
