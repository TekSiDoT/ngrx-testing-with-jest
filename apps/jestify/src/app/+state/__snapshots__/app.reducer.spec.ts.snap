// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Reducer unknown action should return the initial state 1`] = `
Object {
  "jesters": 1,
  "partyHard": false,
}
`;

exports[`App Reducer valid App actions  should properly set MANY jesters 1`] = `
Object {
  "jesters": 1000,
  "partyHard": false,
}
`;

exports[`App Reducer valid App actions  should properly set jesters 1`] = `
Object {
  "jesters": 1,
  "partyHard": false,
}
`;
